pallets/phala/src/stakepool.rs

```diff
@@ -541,6 +557,106 @@ pub mod pallet {
 			Ok(())
 		}
 
+		/// Manually assign rewards to pools by miningSwitchOrigin for fixing issue 763
+		///
+		/// Requires:
+		/// 1. The caller is root
+		/// 2. assigned pool must currently exist
+		/// 3. reeward is positive
+		#[pallet::weight(0)]
+		pub fn force_assign_reward(
+			origin: OriginFor<T>,
+			reward_arr: Vec<(u64, BalanceOf<T>)>,
+		) -> DispatchResult {
+
+			// Origin must be MiningSwitchOrigin
+			T::MiningSwitchOrigin::ensure_origin(origin)?;
+
+			for (pid, reward) in reward_arr {
+			    //The assigned pool must exist
+
+			    let mut pool_info = Self::ensure_pool(pid)?;
+			    //reward must be positive
+			    if reward <= Zero::zero() {
+				continue
+			    }
+			    //assign reward
+			    Self::handle_pool_new_reward(&mut pool_info, reward);
+			    StakePools::<T>::insert(&pid, &pool_info);
+ 
+			}
+			Ok(())
+		}
+		
+		/// Claims pool-owner's pending rewards of the sender and send to the `target`
+		/// The rewards associate to sender's "staker role" will not be claimed
+		/// Requires:
+		/// 1. The sender is a pool owner
+		#[pallet::weight(0)]
+		pub fn claim_owner_rewards(
+			origin: OriginFor<T>,
+			pid: u64,
+			target: T::AccountId,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			let mut pool_info = Self::ensure_pool(pid)?;
+			let mut rewards = BalanceOf::<T>::zero();
```

Comment:
Not necessary to make it `mut` because it's just one-time used

Reply:
done